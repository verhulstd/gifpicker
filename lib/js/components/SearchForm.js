// @flow
import * as React from 'react';
import { connect } from 'react-redux';
import {
  setInput,
  getSuggestions,
  clearSuggestions,
} from '../actions/formActions';
import { getGifs } from '../actions/resultActions';

type Props = {
  dispatchInput: Function,
  dispatchSubmit: Function,
  inputValue: string,
  chosen: string,
};
type State = {};

export class SearchForm extends React.Component<Props, State> {
  render() {
    const { dispatchInput, chosen, dispatchSubmit, inputValue } = this.props;
    return (
      <>
        {chosen === '' && (
          <form action="" onSubmit={dispatchSubmit}>
            <button type="submit" className="gifpicker__submit"></button>
            <input
              autoFocus
              placeholder="input search text"
              onChange={dispatchInput}
              value={inputValue}
              className="gifpicker__input"
            />
          </form>
        )}
      </>
    );
  }
}
export const mapStateToProps = state => ({
  apikey: state.settings.apikey,
  inputValue: state.form.input,
  chosen: state.result.chosen,
});
export const mapDispatchToProps = dispatch => ({
  dispatchInput: e => {
    const { value } = e.target;
    dispatch(setInput(value));
    if (value.length >= 3) {
      dispatch(getSuggestions());
    } else {
      dispatch(clearSuggestions());
    }
  },
  dispatchSubmit: e => {
    e.preventDefault();
    dispatch(getGifs());
  },
});
export default connect<*, *, *, *, *, *>(
  mapStateToProps,
  mapDispatchToProps
)(SearchForm);
