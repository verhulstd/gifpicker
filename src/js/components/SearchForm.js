// @flow
import * as React from 'react';
import { connect } from 'react-redux';
import { Input } from 'antd';
import {
  setInput,
  getSuggestions,
  clearSuggestions,
} from '../actions/formActions';
import { getGifs } from '../actions/resultActions';

const { Search } = Input;

type Props = {
  dispatchInput: Function,
  dispatchSubmit: Function,
  inputValue: string,
  chosen: string,
};
type State = {};

class SearchForm extends React.Component<Props, State> {
  render() {
    const { dispatchInput, chosen, dispatchSubmit, inputValue } = this.props;
    return (
      <>
        {chosen === '' && (
          <Search
            autoFocus
            placeholder="input search text"
            onSearch={dispatchSubmit}
            onChange={dispatchInput}
            style={{ width: '100%' }}
            value={inputValue}
            className="inputField"
          />
        )}
      </>
    );
  }
}
const mapStateToProps = state => ({
  apikey: state.settings.apikey,
  inputValue: state.form.input,
  chosen: state.result.chosen,
});
const mapDispatchToProps = dispatch => ({
  dispatchInput: e => {
    const { value } = e.target;
    dispatch(setInput(value));
    if (value.length >= 3) {
      dispatch(getSuggestions());
    } else {
      dispatch(clearSuggestions());
    }
  },
  dispatchSubmit: e => {
    dispatch(setInput(e));
    dispatch(getGifs());
  },
});
export default connect<*, *, *, *, *, *>(
  mapStateToProps,
  mapDispatchToProps
)(SearchForm);
