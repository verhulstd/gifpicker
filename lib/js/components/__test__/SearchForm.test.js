import React from 'react';
import { shallow } from 'enzyme';
import { SearchForm, mapStateToProps, mapDispatchToProps } from '../SearchForm';

describe('SearchForm', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<SearchForm />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should call onSubmit when form is submitted', () => {
    const onClick = jest.fn().mockName('onClick');
    const wrapper = shallow(<SearchForm dispatchSubmit={onClick} chosen="" />);
    wrapper.find('form').simulate('submit');
    expect(onClick).toHaveBeenCalled();
  });
  it('should call onChange when input gets changed', () => {
    const onChange = jest.fn().mockName('onClick');
    const wrapper = shallow(<SearchForm dispatchInput={onChange} chosen="" />);
    wrapper.find('.gifpicker__input').simulate('change');
    expect(onChange).toHaveBeenCalled();
  });
  it('mapStateToProps should return the right value', () => {
    const initialState = {
      settings: { apikey: '123' },
      form: { input: '456' },
      result: { chosen: '789' },
    };
    expect(mapStateToProps(initialState).chosen).toEqual('789');
  });
  it('should call dispatch method dispatchInput correctly', () => {
    const dispatch = jest.fn();
    const mockedEvent = Object.assign(jest.fn(), {
      preventDefault: () => {},
      target: { value: 'test' },
    });
    mapDispatchToProps(dispatch).dispatchInput(mockedEvent);
    expect(dispatch.mock.calls.length).toEqual(2);
  });
  it('should call dispatch method dispatchSubmit correctly', () => {
    const dispatch = jest.fn();
    const mockedEvent = Object.assign(jest.fn(), {
      preventDefault: () => {},
    });
    mapDispatchToProps(dispatch).dispatchSubmit(mockedEvent);
    expect(dispatch.mock.calls.length).toEqual(1);
  });
});
