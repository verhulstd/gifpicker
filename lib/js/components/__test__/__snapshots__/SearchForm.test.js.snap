// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should match snapshot 1`] = `ShallowWrapper {}`;
